### ISSUE TITLE
Update to latest geth

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#6317]
In the latest `geth` releases there were some changes related to error handling.

- `geth` relase notes
https://github.com/ethereum/go-ethereum/releases

 To be done after #6423 
- Test PR:
https://app.circleci.com/pipelines/github/raiden-network/raiden/9259/workflows/ca8cbe5d-1d9c-4e74-9df9-65a4dc6b8256/jobs/127897/parallel-runs/23?filterBy=FAILED
- new PR https://github.com/raiden-network/raiden/pull/6426
